soccer-records-model/src/main/java/cz/muni/fi/pa165/team/TeamMatchService.java

```diff
@@ -0,0 +1,195 @@
+package cz.muni.fi.pa165.team;
+
+import cz.muni.fi.pa165.team.exceptions.GoalWithSameParametersAlreadyExistsException;
+import cz.muni.fi.pa165.team.exceptions.MatchTimeCollisionException;
+import cz.muni.fi.pa165.team.exceptions.MatchWithSameParametersAlreadyExistsException;
+import org.springframework.util.Assert;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Tomas Smid <smid.thomas@gmail.com>
+ */
+public class TeamMatchService
+{
+
+    /**
+     * Non-arg constructor of class TeamMatchService
+     */
+    public TeamMatchService()
+    {
+
+    }
+
+    /**
+     * This method creates new match if all needed parameters are valid.
+     *
+     * @param allMatchesOfHomeTeam matches in which appears team which should be
+     *                             set up as home team in match which should be created
+     * @param allMatchesOfAwayTeam matches in which appears team which should be
+     *                             set up as away team in match which should be created
+     * @param homeTeam home team of match which should be created
+     * @param awayTeam away team of match which should be created
+     * @param startTime start time of match which should be created
+     * @param endTime end time of match which should be created
+     * @return newly created match
+     */
+    public TeamMatch createMatch(
+        List<TeamMatch> allMatchesOfHomeTeam,
+        List<TeamMatch> allMatchesOfAwayTeam,
+        Team homeTeam,
+        Team awayTeam,
+        Date startTime,
+        Date endTime
+    )
+    {
+        Assert.notNull(homeTeam, "Match cannot be created with a null home team");
+        Assert.notNull(awayTeam, "Match cannot be created with a null away team");
+        Assert.isTrue(!homeTeam.equals(awayTeam), "Match cannot be created for home and away teams which are same");
+
+        validateMatchTimes(startTime, endTime);
+
+        if (!allMatchesOfHomeTeam.isEmpty()) {
```

Comment:
pokud to necháš jako kolekce, tak tyhle dva cykly bych zrefaktoroval do metody a tu zavolal dvakrát pro ty jednotlivé kolekce